presto-main/src/main/java/com/facebook/presto/operator/window/WindowPartition.java

```diff
@@ -176,23 +176,34 @@ else if (frameInfo.getType() == RANGE) {
 
     private boolean emptyFrame(int rowPosition, int endPosition)
     {
-        if (frameInfo.getStartType() != frameInfo.getEndType()) {
+        FrameBound.Type startType = frameInfo.getStartType();
+        FrameBound.Type endType = frameInfo.getEndType();
+
+        int positions = endPosition - rowPosition;
+
+        if (startType == UNBOUNDED_PRECEDING && endType == PRECEDING) {
+            return getEndValue() > rowPosition;
+        }
+
+        if (startType == FOLLOWING && endType == UNBOUNDED_FOLLOWING) {
+            return getStartValue() > positions;
+        }
+
+        if (startType != endType) {
             return false;
         }
 
-        FrameBound.Type type = frameInfo.getStartType();
```

Comment:
Let's keep this variable `type` to make it clear that at this point, the frames have the same type